File: Sources/Frontend/src/rendering/components/models/ActionData/TilePlacementData.ts
import { GridPositionComponent } from "@/rendering/components/GridPosiotionComponent";
import { IActionData } from "./IActionData";
import { PassageType } from "../PassageType";

export class TilePlacementData extends IActionData {
   constructor(
      player: string,
      public type: PassageType = PassageType.FourWay,
      public to: GridPositionComponent = new GridPositionComponent(0,0),
      public rotation: number = 0
   ) {
      super();
      this.playerId = player;
   }
}

File: Sources/Backend/TheLastCandle/Models/ActionData/TilePlacementData.cs
using TheLastCandle.Models.Components;
using TheLastCandle.Models.Events;

namespace TheLastCandle.Models.ActionData
{
    public class TilePlacementData : IClientActionData
    {
        public PassageType type { get; set; }
        public GridPosition to { get; set; }
        public int rotation { get; set; }
    }
}
